bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/commands/CommandImagePersistence.java

```diff
@@ -77,8 +85,6 @@ public final class CommandImagePersistence extends RegistryPersistence {
 	private static void readImagesFromRegistry(final IConfigurationElement[] configurationElements,
 			final int configurationElementCount, final CommandImageManager commandImageManager,
 			final ICommandService commandService) {
-		// Undefine all the previous images.
-		commandImageManager.clear();
```

Comment:
The method can be called multiple times :

![image](https://user-images.githubusercontent.com/208716/227856358-3d077106-d808-4ca7-881f-ea94b4d27db8.png)

So I'm not sure that removing this is okay nor why it was done.

Reply:
I did not remove it, I just moved it up to the method that calls both readImages functions. 
So that it makes more sense from a flow perspective.

Previously it was:
read() -> readImagesFromRegistry() -> clear()

Now it is:
read() -> clear()
read() -> readImagesFromRegistry()
read() ->  readImagesFromApplicationModel()

Reply:
Sorry for misreading.